src/meta/expr.lean

```diff
@@ -307,6 +307,14 @@ e.fold mk_name_set (λ e' _ es, if e'.is_constant then es.insert e'.const_name e
 meta def list_meta_vars (e : expr) : list expr :=
 e.fold [] (λ e' _ es, if e'.is_mvar then insert e' es else es)
 
+/--
+Test if an expression `e` "may occur" in `t`, in the sense that either
+it does appear in `t`, or `t` contains metavariables.
+-/
+meta def may_occur (e : expr) (t : expr) : tactic unit :=
```

Comment:
This doesn't need to be `tactic` valued, does it? I'd prefer the more verbose but more descriptive name `contains_expr_or_mvar` or something like that. (My suggestion would need the order of the arguments to flip, so rephrase as desired.)

Reply:
Fixed both suggestions!